crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_assign.rs

```diff
@@ -0,0 +1,214 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, Rule, RuleDiagnostic,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsAssignment, AnyJsExpression, JsAssignmentExpression, JsAssignmentOperator,
+    JsBinaryExpression, JsBinaryOperator, JsSyntaxKind, T,
+};
+use biome_rowan::{AstNode, BatchMutationExt};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Require assignment operator shorthand where possible
+    ///
+    /// JavaScript provides shorthand operators that combine variable assignment and some simple mathematical operations
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/operator-assignment/
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
```

Comment:
I think we can remove the variable declaration. This allows focusing on the actual change of the rule.
```suggestion
```

Same for the other examples.

Reply:
Sure